File: DecompiledGameScripts/core/gameplay/prereqs/damage/hitFlagHitPrereqCondition.swift

public class HitFlagHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_hitFlag: hitFlag;

  public let m_invertHitFlag: Bool;

  public func SetData(recordID: TweakDBID) -> Void {
    let str: String = TweakDBInterface.GetString(recordID + t".hitFlag", "");
    this.m_hitFlag = IntEnum<hitFlag>(Cast<Int32>(EnumValueFromString("hitFlag", str)));
    this.m_invertHitFlag = TweakDBInterface.GetBool(recordID + t".invertHitFlag", false);
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let result: Bool = hitEvent.attackData.HasFlag(this.m_hitFlag);
    if this.m_invertHitFlag {
      result = !result;
    };
    if result {
      result = this.CheckOnlyOncePerShot(hitEvent);
    };
    return this.m_invert ? !result : result;
  }
}

File: Definitions/Game_Definitions/classes/HitFlagHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class HitFlagHitPrereqCondition : BaseHitPrereqCondition
---@field hitFlag hitFlag
---@field invertHitFlag Bool
HitFlagHitPrereqCondition = {}

---@return HitFlagHitPrereqCondition
function HitFlagHitPrereqCondition.new() return end

---@param props table
---@return HitFlagHitPrereqCondition
function HitFlagHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function HitFlagHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function HitFlagHitPrereqCondition:SetData(recordID) return end

